src/simplify_reshapes.cpp

```diff
@@ -806,6 +806,28 @@ struct find_transpose_slice
     }
 };
 
+// replace scalar (multibroadcast->unsqueeze) with just an updated (multibroadcast)
+struct find_scalar_multibroadcast_unsqueeze
+{
+    auto matcher() const
+    {
+        return match::name("unsqueeze")(match::arg(0)(match::name("multibroadcast")));
+    }
+
+    void apply(module& m, const match::matcher_result& mr) const
+    {
+        auto unsq      = mr.result;         // unsqeeze
+        auto mbr       = unsq->inputs()[0]; // multibroadcast
+        auto mbr_shape = mbr->get_shape();
+        if(mbr_shape.ndim() != 1 and mbr_shape.strides()[0] != 0)
+            return; // ignore except multi-broadcasted scalers
```

Comment:
nit: if it's multibroadcasted with ndim==1 then strides are guranteed to be zero. therefore you only need to check for ndim==1.